index.js

```diff
@@ -1,71 +1,109 @@
+    }
+    ResHandler.resultHandler(resp)
+    return callback(new Error('Kinesis stream failed or contained no records'))
+  } else {
+    await exports.parseRecords(records)
+    return callback(null, 'Successfully parsed records')
+  }
+}
 
-exports.handler = (event, context, callback) => {
-    records = event['Records']
-    for(var i = 0; i < records.length; i++){
-        record = records[i]
-        let payload = JSON.parse(new Buffer.from(record.kinesis.data, 'base64').toString('ascii'))
-        let url = payload['url']
-        let fileNameMatch = fileNameRegex.exec(url)
-        if (!fileNameMatch){
-            handleResp = {
-                "status": 500,
-                "code": "Regex Failure",
-                "message": "Failed to extract file from url " + url
-            }
-            resultHandler(handleResp)
-            continue
-        }
-        let fileName = fileNameMatch[0]
-        let itemID = payload['id']
-        let updated = new Date(payload['updated'])
-        checkForExisting(fileName, updated).then((status) => {
-            axios({
-                method: 'get',
-                url: url,
-                responseType: 'stream'
-            })
-            .then((response) => {
-                epubExplode(fileName, itemID, updated, response)
-                getBuffer(response.data).then((buffer) => {
-                    epubStore(fileName, itemID, updated, 'archive', buffer)
-                })
-                .catch((error) => {
-                    handleResp = {
-                        "status": 500,
-                        "code": "Stream-to-Buffer Error",
-                        "message": error
-                    }
-                    resultHandler(handleResp)
-                })
+exports.parseRecords = async (records) => {
+  let results = records.map(exports.parseRecord)
+  return new Promise((resolve, reject) => {
+    Promise.all(results).then((responses) => {
+      responses.forEach((resp) => {
+        ResHandler.resultHandler(resp)
+      })
+      resolve()
+    })
+  })
+}
+
+exports.runAccessCheck = async (zipData, itemID) => {
+  try{
+    let accessReport = await AccessibilityChecker.getAccessibilityReport(zipData)
+    accessReport['id'] = itemID
+    return {
+      "status": 200,
+      "code": "accessibility",
+      "message": "Created Accessibility Score",
+      "data": accessReport
+    }
+  } catch(err) {
+    return {
+      "status": 500,
+      "code": "Accessibility Report Error",
+      "data": {
+        "id": itemID
+      },
+      "message": JSON.stringify(err, Object.getOwnPropertyNames(err))
+    }
+  }
+}
 
-            })
-            .catch((error) => {
-                handleResp = {
-                    "status": error.response.status,
-                    "code": "Axios Failure",
-                    "message":error.response.data
-                }
-                resultHandler(handleResp)
-            })
+exports.parseRecord = (record) => {
```

Comment:
This feels like two functions to me. One to parse the JSON, check the filename, and either throw an error or return the filename, id, and date; and another to take those and do the existence checking and exploding.

Reply:
I agree, I've split out the functionality into two separate methods. It revealed a few issues with the code and made the test suite coverage much more accurate